extensions/ql-vscode/src/databases.ts

```diff
@@ -470,6 +475,25 @@ export class DatabaseItemImpl implements DatabaseItem {
     return uri.scheme === zipArchiveScheme &&
       decodeSourceArchiveUri(uri).sourceArchiveZipPath === this.sourceArchive.fsPath;
   }
+
+  public async isAffectedByTest(testPath: string): Promise<boolean> {
+    const databasePath = this.databaseUri.fsPath;
+    if (!databasePath.endsWith('.testproj')) {
+      return Promise.resolve(false);
+    }
+    const stats = await fs.stat(testPath);
```

Comment:
Does `stat` return falsy if the path doesn't exist or does it throw?

Reply:
Good catch!  I ran a test, and it throws if the path does not exist.  I updated the code and the tests accordingly.